vtm/src/org/oscim/map/ViewController.java

```diff
@@ -86,12 +86,12 @@ public void setMapScreenCenter(float pivotY) {
      * @param mx the amount of pixels to move the map horizontally.
      * @param my the amount of pixels to move the map vertically.
      */
-    public void moveMap(float mx, float my) {
+    public synchronized void moveMap(float mx, float my) {
         ThreadUtils.assertMainThread();
 
-        Point p = applyRotation(mx, my);
+        applyRotation(mx, my, mPos.bearing, mMovePoint);
         double tileScale = mPos.scale * Tile.SIZE;
-        moveTo(mPos.x - p.x / tileScale, mPos.y - p.y / tileScale);
```

Comment:
`mPos` and `p` (reference to object) could have been changed while other `moveMap` calls are done. If I interpret that right.